syntax ="proto3";

message VgaState {
  uint32  misc_output             = 1;
  uint32  status                  = 2;
  uint32  sequencer_index         = 3;
  bytes   sequencer               = 4;
  uint32  gfx_index               = 5;
  bytes   gfx                     = 6;
  uint32  attribute_index         = 7;
  bytes   attribute               = 8;
  uint32  crtc_index              = 9;
  bytes   crtc                    = 10;
  uint32  palette_read_index      = 11;
  uint32  palette_write_index     = 12;
  bytes   palette                 = 13;
  uint32  dac_state               = 14;
  uint32  feature_control         = 15;
  uint32  latch                   = 16;
}

message VbeState {
  uint32  version                 = 1;
  uint32  index                   = 2;
  bytes   registers               = 3;
}
